x/paramfilter/test/param_filter_test.go

```diff
@@ -1,63 +1,119 @@
 package test
 
 import (
+	"fmt"
 	"testing"
 
 	"github.com/celestiaorg/celestia-app/app"
 	testutil "github.com/celestiaorg/celestia-app/test/util"
 	"github.com/celestiaorg/celestia-app/x/paramfilter"
+	"github.com/cosmos/cosmos-sdk/baseapp"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/x/params/types/proposal"
 	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
 	"github.com/stretchr/testify/require"
 	tmlog "github.com/tendermint/tendermint/libs/log"
-	"github.com/tendermint/tendermint/proto/tendermint/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	coretypes "github.com/tendermint/tendermint/types"
 )
 
 func TestParamFilter(t *testing.T) {
-	app, _ := testutil.SetupTestAppWithGenesisValSet(app.DefaultConsensusParams())
-
-	require.Greater(t, len(app.BlockedParams()), 0)
+	testApp, _ := testutil.SetupTestAppWithGenesisValSet(app.DefaultConsensusParams())
+	require.Greater(t, len(testApp.BlockedParams()), 0)
 
 	// check that all blocked parameters are in the filter keeper
-	pph := paramfilter.NewParamBlockList(app.BlockedParams()...)
-	for _, p := range app.BlockedParams() {
+	pph := paramfilter.NewParamBlockList(testApp.BlockedParams()...)
+	for _, p := range testApp.BlockedParams() {
 		require.True(t, pph.IsBlocked(p[0], p[1]))
 	}
 
-	handler := pph.GovHandler(app.ParamsKeeper)
-	ctx := sdk.NewContext(app.CommitMultiStore(), types.Header{}, false, tmlog.NewNopLogger())
+	handler := pph.GovHandler(testApp.ParamsKeeper)
+	ctx := sdk.NewContext(testApp.CommitMultiStore(), tmproto.Header{}, false, tmlog.NewNopLogger())
+
+	t.Run("test that a proposal with a blocked param is rejected", func(t *testing.T) {
+		for _, p := range testApp.BlockedParams() {
+			p := testProposal(proposal.NewParamChange(p[0], p[1], "value"))
+			err := handler(ctx, p)
+			require.Error(t, err)
+			require.Contains(t, err.Error(), "parameter can not be modified")
+		}
+	})
+
+	t.Run("test that a proposal with another blocked param (validator consensus params) is rejected", func(t *testing.T) {
```

Comment:
[Question] Can you please elaborate how this test differs from the first test? Thanks

Reply:
The first test uses the first blocked param from the list of blocked params.

The second test (this test) uses a different blocked param. Specifically the validator consensus params.

Reply:
> The first test uses the first blocked param from the list of blocked params.

In the first test, it looks like it ranges through all of the blocked params, doesn't it? 
```go
                for _, p := range testApp.BlockedParams() {
			p := testProposal(proposal.NewParamChange(p[0], p[1], "value"))
			err := handler(ctx, p)
			require.Error(t, err)
			require.Contains(t, err.Error(), "parameter can not be modified")
		}
```

Reply:
oh true! I can remove the second test case b/c it should be covered by test case one.

Reply:
Resolved by f0a9951bd65a3e9ba6bbc265ab1bd3d255a5e212